File: app-mvvm/src/main/java/com/xuie/imaginary/gui/gank/meizhi/MeizhiContract.java
package com.xuie.imaginary.gui.gank.meizhi;

import com.xuie.imaginary.BasePresenter;
import com.xuie.imaginary.BaseView;
import com.xuie.imaginary.data.BaseBean;

import java.util.List;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface MeizhiContract {

    interface View extends BaseView {
        /**
         * 添加妹子列表
         *
         * @param isRefresh true : 重新刷新； false : 加载
         * @param meiZhis   the mei zhis
         */
        void addList(boolean isRefresh, List<BaseBean> meiZhis);
    }

    interface Presenter extends BasePresenter {
        /**
         * 获取列表
         *
         * @param isRefresh the is refresh
         */
        void getList(boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/gank/meizhi/MeizhiContract.kt
package com.xuie.imaginaryandroid.gui.gank.meizhi

import com.xuie.imaginaryandroid.BasePresenter
import com.xuie.imaginaryandroid.BaseView
import com.xuie.imaginaryandroid.data.BaseBean

/**
 * This specifies the contract between the view and the presenter.
 */
interface MeizhiContract {

    interface View : BaseView<Presenter> {

        fun addList(isRefresh: Boolean, meiZhis: List<BaseBean>)
    }

    interface Presenter : BasePresenter {
        fun getList(isRefresh: Boolean)
    }
}
